plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -506,10 +526,168 @@ export class PersonState {
 
             const deletePersonMessages = await this.db.deletePerson(otherPerson, client)
 
-            return [[...updatePersonMessages, ...distinctIdMessages, ...deletePersonMessages], person]
+            let personOverrideMessages: ProducerRecord[] = []
+            if (this.poEEmbraceJoin) {
+                personOverrideMessages = [await this.addPersonOverride(otherPerson, mergeInto, client)]
+            }
+
+            return [
+                [...personOverrideMessages, ...updatePersonMessages, ...distinctIdMessages, ...deletePersonMessages],
+                person,
+            ]
         })
     }
 
+    private async addPersonOverride(
+        oldPerson: Person,
+        overridePerson: Person,
+        client?: PoolClient
+    ): Promise<ProducerRecord> {
+        const mergedAt = DateTime.now()
+        const oldestEvent = overridePerson.created_at
+        /**
+            We'll need to do 4 updates:
+
+         1. Add the persons involved to the helper table (2 of them)
+         2. Add an override from oldPerson to override person
+         3. Update any entries that have oldPerson as the override person to now also point to the new override person
+
+            TODO: how do we want to be updating oldest_event?
+
+         I'm thinking: we write it if it's not there, but don't update it on conflicts (alternative update to the older date)
+         In the transitive one the same don't update or update to older data (if that's not too complex to do)
+         it's an optimization anyway and squash job might be updating those values anyway and it's just a hint, so ignoring if complex seems better
+         in any case it's important that we use the oldest_event we stored in postgres in CH too.
+         */
+        const oldPersonId = await this.addPersonOverrideHelper(oldPerson, client)
+        const overridePersonId = await this.addPersonOverrideHelper(overridePerson, client)
+
+        const {
+            rows: [{ version }],
+        } = await this.db.postgresQuery(
+            SQL`
+                INSERT INTO posthog_personoverride (
+                    team_id,
+                    old_person_id,
+                    override_person_id,
+                    oldest_event,
+                    version
+                ) VALUES (
+                    ${this.teamId},
+                    ${oldPersonId},
+                    ${overridePersonId},
+                    ${oldestEvent},
+                    1
```

Comment:
why not 0?

Reply:
This commit was cherry-picked from your PR here: https://github.com/PostHog/posthog/pull/14368/files#diff-c51e19d65898f9cd1b4aaad32e5163e7b44507026ce49301507a2cbd6fe5ccdbR572

Which originally came from @hazzadous PR? Not sure about this.

Regardless, my point is I have no idea why 1 here and not 0. So, I can just do that change.

Reply:
Made it 0.